nuclear/extension/NUClearExtension.cmake

```diff
@@ -30,5 +30,14 @@ if(NUCLEAR_LINK_TYPE STREQUAL "SHARED")
   set_target_properties(nuclear_extension PROPERTIES LIBRARY_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}/bin/lib")
 endif()
 
+# Add warnings for the extensions. Use the default ROLE warnings setting if it's not configured manually
+if(NOT DEFINED NUCLEAR_ROLE_WARNINGS)
```

Comment:
I'm not 100% sold on the whole NUCLEAR_ROLE name, because there are things called roles in here that could have warnings. Maybe change it to NUCLEAR_ROLES?

Reply:
I changed the global one to NUCLEAR_ROLES :+1: